cpp/src/interop/dlpack.cpp

```diff
@@ -161,6 +161,16 @@ std::unique_ptr<table> from_dlpack(DLManagedTensor const* managed_tensor,
     CUDF_EXPECTS(tensor.shape[1] < std::numeric_limits<size_type>::max(),
                  "DLTensor second dim exceeds size supported by cudf");
   }
+  if (tensor.ndim == 1) {
+    CUDF_EXPECTS(nullptr == tensor.strides || tensor.strides[0] == 1,
+                 "from_dlpack of 1D tensor only for unit-stride data");
+  } else if (tensor.ndim == 2) {
+    CUDF_EXPECTS(
+      nullptr != tensor.strides && tensor.strides[0] == 1 && tensor.strides[1] == tensor.shape[0],
+      "from_dlpack of 2D tensor only for column-major unit-stride data");
+  } else {
+    CUDF_UNREACHABLE("Unhandled tensor dimension in from_dlpack");
```

Comment:
```suggestion
    CUDF_FAIL("Unhandled tensor dimension in from_dlpack");
```
I think `CUDF_FAIL` is more appropriate here.

https://github.com/rapidsai/cudf/blob/13c21d02a9c44b0f5b6c7ecb981272ec9440a01c/cpp/include/cudf/detail/utilities/assert.cuh#L42-L45

Reply:
Thanks.

Reply:
Maybe these should be combined. Remove the initial check and change the `else` statement here to
```
} else {
CUDF_FAIL("DLTensor must be 1D or 2D")
}
```